components/chef-ui-library/src/molecules/chef-phat-radio/chef-phat-radio.tsx

```diff
@@ -32,17 +35,24 @@ export class ChefPhatRadio {
 
   @Listen('click') handleClick(event) {
     const option = event.target.closest('chef-option');
+    // If new click option is the same as current, we are deselecting
+    const isSameOption = this.value === option.value;
+
     if (option) {
-      this.value = option.value;
-      this.change.emit();
-      this.input.emit();
+    this.value = this.canDeselect && isSameOption ? '' : option.value;
+    this.change.emit();
+    this.input.emit();
     }
   }
 
   componentDidLoad() {
     const options = this.clearOptions();
     this.selected = find(['value', this.value], options);
```

Comment:
I might be following the logic wrong but seems like the rest of this method could be simplified to:

```ts
if (!this.selected && !this.canDeselect) {
  this.selected = this.el.querySelector('chef-option[selected]') || options[0];
  this.value = this.selected.value;
  this.selected.selected = true;
}


```

Reply:
you are totally right - thanks!

looks like `this.selected.selected = true` needs to stay outside the brackets though.

Reply:
Now I remember why i did this.  When the radio group is deselectable, and doesn't have a value, we can't mark `this.selected.selected` as `true`.  If we move that call inside the brackets, then it doesn't get marked as true when a value is passed in.

I updated this to read a cleaner (in my opinion) to a gate clause.